kura/org.eclipse.kura.linux.position/src/main/java/org/eclipse/kura/linux/position/GpsdPositionProvider.java

```diff
@@ -0,0 +1,300 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eurotech and/or its affiliates and others
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.linux.position;
+
+import static java.lang.Math.toRadians;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.kura.linux.position.GpsDevice.Listener;
+import org.eclipse.kura.position.NmeaPosition;
+import org.osgi.util.measurement.Measurement;
+import org.osgi.util.measurement.Unit;
+import org.osgi.util.position.Position;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.taimos.gpsd4java.api.IObjectListener;
+import de.taimos.gpsd4java.backend.GPSdEndpoint;
+import de.taimos.gpsd4java.types.ATTObject;
+import de.taimos.gpsd4java.types.DeviceObject;
+import de.taimos.gpsd4java.types.DevicesObject;
+import de.taimos.gpsd4java.types.ENMEAMode;
+import de.taimos.gpsd4java.types.SKYObject;
+import de.taimos.gpsd4java.types.TPVObject;
+import de.taimos.gpsd4java.types.subframes.SUBFRAMEObject;
+
+public class GpsdPositionProvider implements PositionProvider, IObjectListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(GpsdPositionProvider.class);
+
+    private GPSdEndpoint gpsEndpoint;
+    private AtomicReference<GpsdInternalState> internalState;
+    private PositionServiceOptions configuration;
+
+    @Override
+    public void start() {
+        if (this.gpsEndpoint != null) {
+            this.gpsEndpoint.start();
+        }
+    }
+
+    @Override
+    public void stop() {
+        if (this.gpsEndpoint != null) {
+            this.gpsEndpoint.stop();
+        }
+    }
+
+    @Override
+    public Position getPosition() {
+        return new Position(
```

Comment:
Should we create a local variable such as
```
final GpsdInternalState currentState = this.internalState.get();
```
and then get the fields out of it? in this way we are sure that all measurements come from the same `TPVObject`.

Reply:
done!